### PR TITLE
SLSA3

### PR DESCRIPTION
**Pull request by rafaelbreno**

### CODE REVIEW

**Review by brooksn — APPROVED** [review]

File: .github/workflows/image-push.yml [hunk 1]
```diff
@@ -23,33 +23,27 @@ jobs:
...
+          secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials password | PRIME_REGISTRY_PASSWORD
 
-    - name: Build container image
-      uses: docker/build-push-action@v5
+    - name: Build and push image
+      uses: rancher/ecm-distro-tools/actions/publish-image@master
       with:
+        image: hardened-kubernetes
+        tag: ${{ github.event.release.tag_name }}
         context: .
```
Comment (brooksn) [inline .github/workflows/image-push.yml:41]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -14,6 +14,27 @@ ifeq ($(TAG),)
...
+	ifeq ($(UNAME_M), x86_64)
+		TARGET_PLATFORMS:=linux/amd64
+	else ifeq ($(UNAME_M), aarch64)
+		TARGET_PLATFORMS:=linux/arm64
+	else 
+		TARGET_PLATFORMS:=linux/$(UNAME_M)
+	endif
+endif
+
+IMAGE ?= $(ORG)/hardened-kubernetes:$(TAG)
```
Comment (brooksn) [inline Makefile:36]:
```suggestion
REPO ?= rancher
IMAGE ?= $(REPO)/hardened-kubernetes:$(shell echo $(TAG) | sed -e 's/+/-/g')
```

File: Makefile [hunk 3]
```diff
@@ -40,7 +61,23 @@ all:
...
+		--sbom=true \
+		--attest type=provenance,mode=max \
+		--platform=$(TARGET_PLATFORMS) \
+		--build-arg PKG=$(PKG) \
+		--build-arg SRC=$(SRC) \
+		--build-arg TAG=$(TAG:$(BUILD_META)=) \
+		--build-arg GO_IMAGE=rancher/hardened-build-base:$(GOLANG_VERSION) \
+		--build-arg K3S_ROOT_VERSION=$(K3S_ROOT_VERSION) \
+		--build-arg K8S_TAG=$(shell echo $(TAG) | grep -oP "^v(([0-9]+)\.([0-9]+)\.([0-9]+))") \
+		--tag $(ORG)/hardened-kubernetes:$(shell echo $(TAG) | sed -e 's/+/-/g') \
```
Comment (brooksn) [inline Makefile:78]:
```suggestion
		--tag $(IMAGE) \
```

**Review by brooksn — CHANGES_REQUESTED** [review]

File: .github/workflows/image-push.yml (hunk 1 above)
Comment (rafaelbreno) [reply .github/workflows/image-push.yml]: Done

File: Makefile (hunk 3 above)
Comment (rafaelbreno) [reply Makefile]: Done

File: Makefile (hunk 2 above)
Comment (rafaelbreno) [reply Makefile]: Done

**Review by brooksn — APPROVED** [review]

**Review by tashima42 — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -14,6 +14,28 @@ ifeq ($(TAG),)
...
+		TARGET_PLATFORMS:=linux/amd64
+	else ifeq ($(UNAME_M), aarch64)
+		TARGET_PLATFORMS:=linux/arm64
+	else 
+		TARGET_PLATFORMS:=linux/$(UNAME_M)
+	endif
+endif
+
+REPO ?= rancher
+IMAGE ?= $(REPO)/hardened-kubernetes:$(shell echo $(TAG) | sed -e 's/+/-/g')
```
Comment (briandowns) [inline Makefile:37]: I think the repo var here is referring to the org and not the repo

File: .github/workflows/image-push.yml [hunk 5]
```diff
@@ -23,33 +23,26 @@ jobs:
...
-        tags: rancher/hardened-kubernetes:${{ github.event.release.tag_name }}
-        file: Dockerfile
-        platforms: linux/amd64,linux/arm64
-        build-args: |
-          TAG=${{ env.TAG }}
-          GO_IMAGE=rancher/hardened-build-base:${{ env.GOLANG_VERSION }}
+        image: hardened-kubernetes
+        tag: ${{ github.event.release.tag_name }}
+        public-repo: rancher
+        public-username: ${{ env.DOCKER_USERNAME }}
```
Comment (nicholasSUSE) [inline .github/workflows/image-push.yml:42]: I think we should handle sensitive data as secrets, env variables can get exposed:

```(yaml)
prime-password: ${{ secrets.PRIME_REGISTRY_PASSWORD }}
```

**Review by nicholasSUSE — CHANGES_REQUESTED** [review]

File: .github/workflows/image-push.yml (hunk 5 above)
Comment (rafaelbreno) [reply .github/workflows/image-push.yml:42]: We're getting those values from vault, the `"Read secrets"` step will retrieve them from vault and set it as `envs`. The Github CI will mask those, so no secrets leak.

File: Makefile (hunk 4 above)
Comment (rafaelbreno) [reply Makefile:36]: Forgot to implement the change to use `$(IMAGE)` everywhere, removed `$(ORG)` as it isn't necessary anymore.

File: .github/workflows/image-push.yml (hunk 5 above)
Comment (brooksn) [reply .github/workflows/image-push.yml:42]: This data is only available through vault, and not with github secrets. This use of read-vault-secrets is pretty much standardized.

Comment (tashima42) [reply .github/workflows/image-push.yml:42]: The EIO vault doesn't use gh secrets, just envs, but it still applies the same obfuscation in logs

**Review by nicholasSUSE — APPROVED** [review]

**Pull request merged**